src/QueryBuilder.php

```diff
@@ -211,13 +212,11 @@ private function buildAccessConditions(): string
 
     private function includeAccessData(string $index): bool
     {
-        if (!in_array($index, IndexerInterface::INDEXES_WITH_ACCESS_CONTROL)) {
-            return false;
-        }
-
-        $permissionProvider = $this->getServiceLocator()->get(PermissionInterface::SERVICE_ID);
-
-        return $permissionProvider instanceof ReverseRightLookupInterface;
+        return (new UseAclSpecification())->isSatisfiedBy(
```

Comment:
Will you register this class in DI container?

Reply:
This is a lib, therefore should not have a DI container and I do not want to change taoAdvancedSearch extension just to set these class there. With OpenSearch implementation this library will go to trash and all the ElasticSearch implementation will be moved to taoAdvancedSearch extension (there we will use proper DI, etc).